modules/AlphaWalletFoundation/AlphaWalletFoundation/Types/TokenProviderType.swift

```diff
@@ -146,6 +146,27 @@ public class TokenProvider: TokenProviderType {
         }
     }
 
+    fileprivate static func shouldRetry(_ error: Error) -> Bool {
+        if let web3Error = error as? Web3.Web3Error {
```

Comment:
Do you want to see if `RpcNodeRetryableRequestError` can be checked here? I'm not sure if `RpcNodeRetryableRequestError` *is* always correct — i.e. if all the enum-cases are indeed retryable (but that's the idea) — though, so better to double check it.

Reply:
not sure about `RpcNodeRetryableRequestError ` here, as its checkes only for APIKit errors and here we can't get it.

Reply:
Hmm.. in that case, how about stub out the body of `shouldRetry()` first , return `true` with a `TODO` and merge? I think we might have to go through and be careful with what is retried or not. We are already retrying anyway, but if we code the error types in, we might think that those are the expected ones.